### ISSUE TITLE
Multiple invocations for compute shaders

### ISSUE THREAD

**Issue by mmoult** [mmoult/SPIRV-Interpreter#3]
Compute shaders may need multiple invocations. It would be best to handle these in parallel which would make barriers much simpler. The difficult part will be diverging control flow and handling multiple data arrays.

Compute shaders should still have 1 input and 1 output despite running multiple invocations. No plans to implement multiple invocations of fragment or vertex shaders since it wouldn't save much work (only parsing of the program I think).

**Comment by mmoult** [comment]
Basic support added with this commit:
https://github.com/mmoult/SPIRV-Interpreter/commit/892f68b23f11903ccbc0980a668070cc6a6abc87

File another ticket with specific features, if necessary.
